src/Glpi/Form/QuestionType/QuestionTypeItemDropdown.php

```diff
@@ -83,4 +103,199 @@ public function getWeight(): int
+        if ($config === null) {
+            return null;
+        }
+
+        return $config->getCategoriesFilter();
+    }
+
+    /**
+     * Retrieve root items ID for the item question type
+     *
+     * @param Question|null $question The question to retrieve the root items ID for
+     * @return int
+     */
+    public function getRootItemsId(?Question $question): int
+    {
+        if ($question === null) {
+            return 0;
+        }
+
+        /** @var ?QuestionTypeItemDropdownExtraDataConfig $config */
+        $config = $this->getExtraDataConfig(json_decode($question->fields['extra_data'], true) ?? []);
+        if ($config === null) {
+            return 0;
+        }
+
+        return $config->getRootItemsId();
+    }
+
+    /**
+     * Retrieve subtree depth for the item question type
+     *
+     * @param Question|null $question The question to retrieve the subtree depth for
+     * @return int
+     */
+    public function getSubtreeDepth(?Question $question): int
+    {
+        if ($question === null) {
+            return 0;
+        }
+
+        /** @var ?QuestionTypeItemDropdownExtraDataConfig $config */
+        $config = $this->getExtraDataConfig(json_decode($question->fields['extra_data'], true) ?? []);
+        if ($config === null) {
+            return 0;
+        }
+
+        return $config->getSubtreeDepth();
+    }
+
+    public function getDropdownRestrictionParams(?Question $question): array
+    {
+        $params   = [];
+        $itemtype = $this->getDefaultValueItemtype($question);
+        if ($question === null || $itemtype === null) {
+            return $params;
+        }
+
+        // Apply categories filter if itemtype is an ITILCategory
+        if (is_a($itemtype, ITILCategory::class, true)) {
+            $categories_filter = $this->getCategoriesFilter($question);
+            if (is_array($categories_filter) && count($categories_filter) > 0) {
+                foreach ($categories_filter as $category) {
+                    $key = match ($category) {
+                        'request'  => 'is_request',
+                        'incident' => 'is_incident',
+                        'problem'  => 'is_problem',
+                        'change'   => 'is_change',
+                        default    => throw new \InvalidArgumentException(
+                            sprintf('Unknown category filter: %s', $category)
+                        ),
+                    };
+
+                    $params[$itemtype::getTableField($key)] = 1;
+                }
+            }
+        }
+
+        // Apply specific root
+        $root_items_id = $this->getRootItemsId($question);
+        if ($root_items_id > 0) {
+            $sons = (new DbUtils())->getSonsOf(
+                $itemtype::getTable(),
+                $root_items_id
+            );
+
+            if (count($sons) > 0) {
+                $params[$itemtype::getTableField('id')] = $sons;
+            }
```

Comment:
`$sons` is not supposed to be empty. Anyways, your code says "if there is the root item has no sons, then we do not even filter the tree".

```suggestion
            $params[$itemtype::getTableField('id')] = $sons;
```

or 

```suggestion
            if (count($sons) > 0) {
                $params[$itemtype::getTableField('id')] = $sons;
            } else {
                $params[$itemtype::getTableField('id')] = $root_items_id;
            }
```